pkg/controller/util/endpoint/controller_utils.go

```diff
@@ -306,16 +306,11 @@ func EndpointsEqualBeyondHash(ep1, ep2 *discovery.Endpoint) bool {
 		return false
 	}
 
-	// Serving and Terminating will only be set when the EndpointSliceTerminatingCondition feature is on.
-	// Ignore their difference if the expected or actual value is nil, which means the feature enablement is changed.
-	// Otherwise all EndpointSlices in the system would be updated on the first controller-manager restart even without
-	// actual changes, leading to delay in processing legitimate updates.
-	// Its value will be set to the expected one when there is an actual change triggering update of this EndpointSlice.
-	if ep1.Conditions.Serving != nil && ep2.Conditions.Serving != nil && *ep1.Conditions.Serving != *ep2.Conditions.Serving {
+	if boolPtrChanged(ep1.Conditions.Serving, ep2.Conditions.Serving) {
```

Comment:
@robscott I wasn't sure about this change -- based on the comments it seems like we didn't use `boolPtrChanged` since we're accounting for feature gate enablement, but seems safe to use `boolPtrChanged` now if we're locking the gate?

Reply:
but you can upgrade from a cluster that has not the feature gate enabled , thus the Conditions.Serving is going to be nil on the endpoints until those are processed again for the CCM, no?

Reply:
ic, so it will be false in that case

Reply:
Yeah, basically if the flag was previously disabled (less common since flag is Beta and on by default now) -- on upgrade to v1.26 the field is now enabled and every EndpointSlice will require an update.

Reply:
seems right to me